tests/FINDOLOGIC/Api/Tests/Responses/Xml21/Xml21ResponseTest.php

```diff
@@ -531,4 +531,25 @@ public function testResponseWillReturnSubItemsOfItemsAsExpected()
 
         $this->assertSame($expectedSubItemDetails, $actualSubItemDetails);
     }
+
+    public function testResponseWillNotBreakIfNoProductsAreFound()
+    {
+        $response = $this->getRealResponseData('demoResponseWithoutProducts.xml');
+
+        $this->assertCount(0, $response->getProducts());
+    }
+
+    public function testResponseWillNotBreakIfNoMainFiltersAreFound()
+    {
+        $response = $this->getRealResponseData('demoResponseWithoutMainFilters.xml');
+
+        $this->assertSame(0, $response->getMainFilterCount());
+    }
+
+    public function testResponseWillNotBreakIfNoOtherFiltersAreFound()
+    {
+        $response = $this->getRealResponseData('demoResponseWithoutOtherFilters.xml');
+
+        $this->assertSame(0, $response->getOtherFilterCount());
```

Comment:
Please also make sure that `$response->getOtherFilters()` returns an empty array and `$response->hasOtherFilters()` returns `false`.